blocks/identity-block/components/SocialSignOn/index.jsx

```diff
@@ -1,100 +1,34 @@
-import React, { useEffect, useState, useRef } from 'react';
+import React from 'react';
 import PropTypes from '@arc-fusion/prop-types';
-import useIdentity from '../Identity';
+import FacebookSignIn from './_children/FacebookSignIn';
+import GoogleSignIn from './_children/GoogleSignIn';
 import './styles.scss';
 
-const SocialSignOn = ({ onError, redirectURL }) => {
-  const { Identity } = useIdentity();
-  const [config, setConfig] = useState(() => Identity?.configOptions ?? {});
-  const [isGoogleInitialized, setIsGoogleInitialized] = useState(false);
-  const [isFacebookInitialized, setIsFacebookInitialized] = useState(false);
-
-  const isGoogleReset = useRef(false);
-
-  useEffect(() => {
-    window.onFacebookSignOn = async () => {
-      try {
-        await Identity.facebookSignOn();
-        window.location = redirectURL;
-      } catch (e) {
-        onError();
-      }
-    };
-  });
-
-  useEffect(() => {
-    const fetchConfig = async () => {
-      await Identity.getConfig();
-      setConfig(Identity.configOptions);
-    };
-    if (Identity && !Identity.configOptions) {
-      fetchConfig();
-    }
-  }, [Identity]);
+import useSocialSignIn from './utils/useSocialSignIn';
 
-  useEffect(() => {
-    const initializeGoogle = async () => {
-      await Identity.initGoogleLogin(null, {
-        width: 300,
-        height: 48,
-        onSuccess: () => {
-          if (!isGoogleReset.current) {
-            isGoogleReset.current = true;
-          } else {
-            window.location = redirectURL;
-          }
-        },
-        onFailure: () => { onError(); },
-      });
-    };
-    if (config.googleClientId && !isGoogleInitialized) {
-      initializeGoogle();
-      setIsGoogleInitialized(true);
-    }
-  }, [Identity,
-    config.googleClientId,
+const SocialSignOn = ({ onError, redirectURL }) => {
+  const {
+    isFacebookInitialized,
     isGoogleInitialized,
-    onError,
-    redirectURL]);
-
-  useEffect(() => {
-    const initializeFacebook = async () => {
-      await Identity.initFacebookLogin(null);
-    };
-    if (config.facebookAppId && !isFacebookInitialized) {
-      initializeFacebook();
-      setIsFacebookInitialized(true);
-    }
-  }, [Identity,
-    config.facebookAppId,
-    isFacebookInitialized]);
+    facebookAppId,
+    googleClientId,
+  } = useSocialSignIn(redirectURL, onError);
 
-  if (!Identity) {
```

Comment:
hmm maybe this identity check for the reason of the google button error on sign in 🤔